extensions/ftp/src/main/java/org/mule/extension/ftp/internal/sftp/connection/SftpClient.java

```diff
@@ -211,14 +212,15 @@ private void configureSession(String user) throws JSchException
 
     private void configureHostChecking(Properties hash) throws JSchException
     {
-        if (getKnownHostsFile() == null)
+        if (knownHostsFile == null)
         {
             hash.put(STRICT_HOST_KEY_CHECKING, "no");
         }
         else
         {
+            checkExists(knownHostsFile);
             hash.put(STRICT_HOST_KEY_CHECKING, "ask");
-            jsch.setKnownHosts(getKnownHostsFile().getAbsolutePath());
+            jsch.setKnownHosts(knownHostsFile);
```

Comment:
If the hosts file is to be an absolute path then so should the identity one, so lines 173 and 177 should be changed.